### PR TITLE
Added properties to disable autoconfiguration for publishing and consuming events to/from Kafka

### PR DESCRIPTION
**Pull request by aupodogov**
This MR add ability to disable autoconfiguration publishing and consuming events to/from Kafka via properties `axon.kafka.publisher.enabled` and `axon.kafka.fetcher.enabled`

### CODE REVIEW

File: kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java [hunk 1]
```diff
@@ -214,11 +214,24 @@ private static String resourceToPath(Resource resource) {
      */
     public static class Publisher {
 
+        /**
+         * Enables autoconfiguration for publishing events to Kafka
```
Comment (smcvb) [inline kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java:218]: Minor change suggestion in the description. Especially stating the default may be helpful to end-users.
```suggestion
         * Enables auto-configuration for publishing events to Kafka. Defaults to {@code true}.
```

File: kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java [hunk 2]
```diff
@@ -460,6 +473,11 @@ public void setEventProcessorMode(EventProcessorMode eventProcessorMode) {
      */
     public static class Fetcher {
 
+        /**
+         * Enables autoconfiguration for consuming events from Kafka
```
Comment (smcvb) [inline kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java:477]: Minor change suggestion in the description. Especially stating the default may be helpful to end-users.
```suggestion
         * Enables auto-configuration for consuming events from Kafka. Defaults to {@code true}.
```

File: kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfiguration.java [hunk 3]
```diff
@@ -91,7 +91,7 @@ public KafkaAutoConfiguration(KafkaProperties properties) {
 
     @Bean("axonKafkaProducerFactory")
     @ConditionalOnMissingBean
-    @ConditionalOnProperty(prefix = "axon.kafka.publisher", name = "enabled", havingValue = "true", matchIfMissing = true)
```
Comment (smcvb) [inline kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfiguration.java]: I could think of a minor optimization, and that's to *not* create the `kafkaMessageConverter` bean when both publishing and consuming are disabled. Or, we could have a double conditional on this entire auto-config class. Granted, that's a pretty minimal optimization, so not overly pressing. I let you decide whether you want to tackle this, @aupodogov.

**Review by smcvb — APPROVED** [review]
Just three small comments remaining. Other than that, I think this looks good!

File: kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java (hunk 2 above)
Comment (aupodogov) [reply kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java]: Applied this suggestion

File: kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java (hunk 1 above)
Comment (aupodogov) [reply kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/KafkaProperties.java]: Applied this suggestion

File: kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfiguration.java (hunk 3 above)
Comment (aupodogov) [reply kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfiguration.java]: I did this optimization!

**Comment by aupodogov** [comment]
@smcvb can we point this change to 4.5.x to release earlier?

**Pull request merged**